src/test/java/com/meilisearch/integration/SettingsTest.java

```diff
@@ -844,6 +845,51 @@ public void testUpdateDistinctAttributeSettingsUsingNull() throws Exception {
         assertEquals(initialDistinctAttribute, resetDistinctAttribute);
     }
 
+    /** Tests of the pagination setting methods */
+    @Test
+    @DisplayName("Test get pagination settings by uid")
+    public void testGetPaginationSettings() throws Exception {
+        Index index = createIndex("testGetPaginationSettings");
+        Settings initialSettings = index.getSettings();
+        Pagination initialPagination = index.getPaginationSettings();
+
+        assertEquals(initialSettings.getPagination().getMaxTotalHits(), 1000);
+        assertNotNull(initialPagination.getMaxTotalHits());
+    }
+
+    @Test
+    @DisplayName("Test update pagination settings")
+    public void testUpdatePaginationSettings() throws Exception {
+        Index index = createIndex("testUpdatePaginationSettings");
+        Pagination newPagination = new Pagination();
+
+        Integer MaxTotalHitsTypos = 100;
+
+        newPagination.setMaxTotalHits(MaxTotalHitsTypos);
+        index.waitForTask(index.updatePaginationSettings(newPagination).getTaskUid());
+        Pagination updatedPagination = index.getPaginationSettings();
+
+        assertTrue(updatedPagination.getMaxTotalHits() == 100);
+    }
+
+    @Test
+    @DisplayName("Test reset pagination settings")
+    public void testResetPaginationSettings() throws Exception {
+        Index index = createIndex("testResetPaginationSettings");
+
+        Pagination initialPagination = index.getPaginationSettings();
+        Pagination newPagination = new Pagination();
+        index.waitForTask(index.updatePaginationSettings(newPagination).getTaskUid());
+        Pagination updatedPagination = index.getPaginationSettings();
+
+        index.waitForTask(index.resetPaginationSettings().getTaskUid());
+        Pagination paginationAfterReset = index.getPaginationSettings();
+
+        assertTrue(initialPagination.getMaxTotalHits() == 1000);
+        assertTrue(updatedPagination.getMaxTotalHits() == 100);
+        assertTrue(paginationAfterReset.getMaxTotalHits() == 1000);
```

Comment:
```suggestion
        assertEquals(1000, initialPagination.getMaxTotalHits());
        assertEquals(100, updatedPagination.getMaxTotalHits());
        assertEquals(1000, paginationAfterReset.getMaxTotalHits());
```
As above, I find it easier to debug with the `assertEquals` to get a more explicit error message :wink: